index.js

```diff
@@ -20,6 +22,110 @@ app.get('/info', (req, res) => {
   res.send({ serverName: packageData.name, serverVersion: packageData.version });
 });
 
+// route for displaying all products
+app.get('/products/all', (req, res) => {
+
+  var jsonOutputObject = {
+    products : []
+  };
+
+  for(var i = 0; i < productData.products.length; i++) {
+    
+    jsonOutputObject.products.push(productData.products[i])
+
+    // Getting category name for the product
+    for(var j = 0; j < categoryData.categories.length; j++) {
+      if(jsonOutputObject.products[i].categoryId == categoryData.categories[j].id) {
```

Comment:
```suggestion
      if(jsonOutputObject.products[i].categoryId === categoryData.categories[j].id) {
```
See [this SFO answer](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons#359509).

Reply:
I've made the chenges, Thank you :)